graphstorm-processing/graphstorm_processing/graph_loaders/schema_utils.py

```diff
@@ -0,0 +1,170 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This module is used to parse the schema of CSV input files, inferring
+the type of the columns from the type mentioned in the configuration.
+"""
+import logging
+from typing import Sequence, List, Type
+
+from pyspark.sql.types import (
+    StructType,
+    StructField,
+    StringType,
+    DataType,
+    FloatType,
+)
+
+from ..config.config_parser import EdgeConfig, NodeConfig
+from ..config.label_config_base import LabelConfig
+from ..config.feature_config_base import FeatureConfig
+
+
+def parse_edge_file_schema(edge_config: EdgeConfig) -> StructType:
+    """Parses schema of CSV edge file according to edge configuration.
+
+    Parameters
+    ----------
+    edge_config : EdgeConfig
+        The configuration object for the edge type.
+
+    Returns
+    -------
+    StructType
+        A Spark StructType describing the typed schema for the edge files.
+    """
+    # TODO: Will need to add support for relation types
+    edge_fields_list = [
+        StructField(edge_config.src_col, StringType(), False),
+        StructField(edge_config.dst_col, StringType(), False),
+        # StructField(relation_type, StringType(), False),
+    ]
+
+    if edge_config.feature_configs:
+        features_schema_list = _parse_features_schema(edge_config.feature_configs)
+        edge_fields_list.extend(features_schema_list)
+    if edge_config.label_configs:
+        labels_schema_list = _parse_edge_labels_schema(edge_config.label_configs)
+        edge_fields_list.extend(labels_schema_list)
+
+    return StructType(edge_fields_list)
+
+
+def _parse_features_schema(features_objects: Sequence[FeatureConfig]) -> Sequence[StructField]:
+    field_list = []
+    for feature_config in features_objects:
+        feature_type = feature_config.feat_type
+        for feature_col, _ in zip(feature_config.cols, feature_config.feat_name):
+            spark_feature_type = determine_spark_feature_type(feature_type)
+
+            field_list.append(StructField(feature_col, spark_feature_type(), True))
+
+    return field_list
+
+
+def determine_spark_feature_type(feature_type: str) -> Type[DataType]:
+    """Returns a DataType class, depending on the type of feature provided.
+
+    Parameters
+    ----------
+    feature_type : str
+        Name of the feature type, e.g. 'numerical' or 'no-op'
+
+    Returns
+    -------
+    Type[DataType]
+        Corresponding Spark type class.
+
+    Raises
+    ------
+    NotImplementedError
+        In case an unsupported feature_type is provided.
+    """
+    # TODO: Replace with pattern matching after moving to Python 3.10?
+    if feature_type in [
+        "category",
+        "datetime",
```

Comment:
Do we support datetime, multi_category and multi_numerical now?

Reply:
We don't I'll remove these.